soc/nxp/imx/imx8m/adsp/linker.ld

```diff
@@ -412,6 +412,13 @@ SECTIONS
     . = ALIGN(4096);
   } >sdram0 :sdram0_phdr
 
+  NonCacheable : ALIGN(4)
```

Comment:
No need to create a new section, this is enough:
```
diff --git a/soc/nxp/imx/imx8m/adsp/linker.ld b/soc/nxp/imx/imx8m/adsp/linker.ld
index 04e17768010..6feb6a99b29 100644
--- a/soc/nxp/imx/imx8m/adsp/linker.ld
+++ b/soc/nxp/imx/imx8m/adsp/linker.ld
@@ -408,6 +408,7 @@ SECTIONS
     _trace_ctx_end = ABSOLUTE(.);
     . = ALIGN(4);
     *(.gna_model)
+    *(NonCacheable)
     __data_end = ABSOLUTE(.);
     . = ALIGN(4096);
   } >sdram0 :sdram0_phdr
```

Add  `*(NonCacheable)` exactly at the end of the `.data` section above.